From akpm@osdl.org  Fri Jun 30 16:41:19 2006
Message-Id: <200606300934.k5U9YlDV001793@shell0.pdx.osdl.net>
From: Christoph Lameter <clameter@sgi.com>
Subject: USB: remove empty destructor from drivers/usb/mon/mon_text.c
To: greg@kroah.com
Cc: akpm@osdl.org, clameter@sgi.com, penberg@cs.helsinki.fi
From: akpm@osdl.org
Date: Fri, 30 Jun 2006 02:34:47 -0700

From: Christoph Lameter <clameter@sgi.com>

Remove destructor and call kmem_cache_create with NULL for the destructor.

Acked-by: Pekka Enberg <penberg@cs.helsinki.fi>
Signed-off-by: Christoph Lameter <clameter@sgi.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Cc: Pete Zaitcev <zaitcev@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/mon/mon_text.c |    7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

--- gregkh-2.6.orig/drivers/usb/mon/mon_text.c
+++ gregkh-2.6/drivers/usb/mon/mon_text.c
@@ -64,7 +64,6 @@ struct mon_reader_text {
 };
 
 static void mon_text_ctor(void *, kmem_cache_t *, unsigned long);
-static void mon_text_dtor(void *, kmem_cache_t *, unsigned long);
 
 /*
  * mon_text_submit
@@ -268,7 +267,7 @@ static int mon_text_open(struct inode *i
 	    (long)rp);
 	rp->e_slab = kmem_cache_create(rp->slab_name,
 	    sizeof(struct mon_event_text), sizeof(long), 0,
-	    mon_text_ctor, mon_text_dtor);
+	    mon_text_ctor, NULL);
 	if (rp->e_slab == NULL) {
 		rc = -ENOMEM;
 		goto err_slab;
@@ -459,7 +458,3 @@ static void mon_text_ctor(void *mem, kme
 	memset(mem, 0xe5, sizeof(struct mon_event_text));
 }
 
-static void mon_text_dtor(void *mem, kmem_cache_t *slab, unsigned long sflags)
-{
-	;
-}
